src/ngx_http_lua_socket_tcp.c

```diff
@@ -4800,48 +5320,29 @@ ngx_http_lua_socket_tcp_setkeepalive(lua_State *L)
     ngx_http_lua_socket_tcp_finalize(r, u);
 #endif
 
+    ngx_http_lua_socket_tcp_resume_conn_op(spool);
```

Comment:
The above `ngx_http_lua_socket_tcp_finalize` already calls `ngx_http_lua_socket_tcp_resume_conn_op` (granted the pool is not empty). Is this second call intentional then?

Reply:
@thibaultcha 
Yes, the call is intentional.
The `ngx_http_lua_socket_tcp_resume_conn_op` won't be called by `ngx_http_lua_socket_tcp_finalize`, because we have set the `u->peer->connection` to `NULL` above. So we need to call it after `ngx_http_lua_socket_tcp_finalize`.
I will add a comment to clarify it.